django_filters/filters.py

```diff
@@ -489,6 +492,102 @@ def __init__(self, *args, **kwargs):
         super(BaseRangeFilter, self).__init__(*args, **kwargs)
 
 
+class OrderingFilter(BaseCSVFilter, ChoiceFilter):
+    """
+    Enable queryset ordering. As an extension of ``ChoiceFilter`` it accepts
+    two additional arguments that are used to build the ordering choices.
+
+    * ``fields`` is a mapping of {model field name: parameter name}. The
+      parameter names are exposed in the choices and mask/alias the field
+      names used in the ``order_by()`` call. Similar to field ``choices``,
+      ``fields`` accepts the 'list of two-tuples' syntax that retains order.
+      ``fields`` may also just be an iterable of strings. In this case, the
+      field names simply double as the exposed parameter names.
+
+    * ``field_labels`` is an optional argument that allows you to customize
+      the display label for the corresponding parameter. It accepts a mapping
+      of {field name: human readable label}. Keep in mind that the key is the
+      field name, and not the exposed parameter name.
+
+    Additionally, you can just provide your own ``choices`` if you require
+    explicit control over the exposed options. For example, when you might
+    want to disable descending sort options.
+
+    This filter is also CSV-based, and accepts multiple ordering params. The
+    default select widget does not enable the use of this, but it is useful
+    for APIs.
```

Comment:
> Was your intention to have the "default select widget" in play?

The select widget needs to be augmented by the `CSVWidget` (similar to how the CSV field and filter work). I'm not sure if this can be handled generically or if this needs to be a one off `CSVSelect` widget.

> I'm thinking BaseCSVField needs to handle that it won't get a list from e.g. a Select widget.

Off the cuff I'd agree - no reason the field couldn't handle both values. That said, a different widget class won't know how to render the list of values it will get from the CSV field. At worst, it raises an exception. At best, it will force the list as a text value. 

I think the correct thing to do here is have conditional rendering in the base CSV widget. If it has a single value (eg, from a select dropdown), use the main widget to render. If it has multiple values, force a text input.